### PR TITLE
[test] Add availability guards for tests checking behavioral changes in 5.1

### PR DESCRIPTION
**Pull request by lorentey**
Add runtime 9999-availability guards around tests checking for behavioral changes since 5.0.

This is useful in case these tests are deployed with the 5.0 stdlib.

rdar://problem/50150948

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci please test

File: test/stdlib/StringBridge.swift [hunk 1]
```diff
@@ -77,6 +77,8 @@ StringBridgeTests.test("Tagged NSString") {
 
 func returnOne<T>(_ t: T) -> Int { return 1 }
 StringBridgeTests.test("Character from NSString") {
+  guard #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) else { return }
```
Comment (lorentey) [inline test/stdlib/StringBridge.swift]: cc @milseman

File: test/stdlib/Inputs/CommonArrayTests.gyb [hunk 2]
```diff
@@ -572,6 +572,11 @@ ${Suite}.test("${ArrayType}/init(unsafeUninitializedCapacity:...:)/validCount")
 }
 
 ${Suite}.test("${ArrayType}/init(unsafeUninitializedCapacity:...:)/reassignBuffer") {
+  guard #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) else {
+    // When back-deployed to 5.0, this coding error does not get detected.
```
Comment (lorentey) [inline test/stdlib/Inputs/CommonArrayTests.gyb]: cc @natecook1000

**Comment by lorentey** [comment]
@swift-ci test

File: test/stdlib/Inputs/CommonArrayTests.gyb (hunk 2 above)
Comment (natecook1000) [reply test/stdlib/Inputs/CommonArrayTests.gyb]: That's… odd. Did I add that check in something that's now part of the ABI? LGTM in any case, thx @lorentey!

Comment (lorentey) [reply test/stdlib/Inputs/CommonArrayTests.gyb]: The buffer reassign check wasn't in 5.0's underscored initializer, so code using the new public initializer won't get this check when it's deployed on macOS 10.14.2 / iOS 12.2 etc. I don't think this is a problem; the missing check won't break any working code.

Our ABI back deployment tests detected this change in behavior between the 5.0 and 5.1 stdlibs; the availability guard I added here acknowledges that this was an expected change.

**Pull request merged**